python/instrumentation/openinference-instrumentation-anthropic/src/openinference/instrumentation/anthropic/_wrappers.py

```diff
@@ -0,0 +1,346 @@
+                            f"{LLM_OUTPUT_MESSAGES}.0.{MESSAGE_CONTENT}": response.completion,
+                            f"{LLM_OUTPUT_MESSAGES}.0.{MESSAGE_ROLE}": "assistant",
+                            OUTPUT_VALUE: response.completion,
+                            OUTPUT_MIME_TYPE: OpenInferenceMimeTypeValues.TEXT,
+                        }
+                    )
+                )
+            )
+
+            span.set_status(trace_api.StatusCode.OK)
+        return response
+
+
+class _AsyncCompletionsGenerationWrapper(_WithTracer):
+    """
+    Wrapper for the pipeline processing
+    Captures all calls to the pipeline
+    """
+
+    async def __call__(
+        self,
+        wrapped: Callable[..., Any],
+        instance: Any,
+        args: Tuple[Any, ...],
+        kwargs: Mapping[str, Any],
+    ) -> Any:
+        if context_api.get_value(context_api._SUPPRESS_INSTRUMENTATION_KEY):
+            return await wrapped(*args, **kwargs)
+
+        llm_invocation_params = kwargs
+        llm_prompt = dict(kwargs).pop("prompt", None)
+
+        # Prepare invocation parameters by merging args and kwargs
+        invocation_parameters = {}
+        for arg in args:
+            if arg and isinstance(arg, dict):
+                invocation_parameters.update(arg)
+        invocation_parameters.update(kwargs)
+
+        span_name = "AsyncCompletions"
+        with self._tracer.start_as_current_span(
+            span_name,
+            record_exception=False,
+            set_status_on_exception=False,
+        ) as span:
+            span.set_attributes(dict(get_attributes_from_context()))
+
+            span.set_attributes(
+                dict(
+                    _flatten(
+                        {
+                            SpanAttributes.OPENINFERENCE_SPAN_KIND: LLM,
+                            SpanAttributes.LLM_INPUT_MESSAGES: llm_prompt,
+                            SpanAttributes.LLM_INVOCATION_PARAMETERS: safe_json_dumps(
+                                llm_invocation_params
+                            ),
+                            SpanAttributes.LLM_MODEL_NAME: llm_invocation_params.get("model"),
+                            SpanAttributes.INPUT_MIME_TYPE: OpenInferenceMimeTypeValues.JSON,
+                        }
+                    )
+                )
+            )
+            try:
+                response = await wrapped(*args, **kwargs)
+            except Exception as exception:
+                span.set_status(trace_api.Status(trace_api.StatusCode.ERROR, str(exception)))
+                span.record_exception(exception)
+                raise
+            span.set_attributes(
+                dict(
+                    _flatten(
+                        {
+                            f"{LLM_OUTPUT_MESSAGES}.0.{MESSAGE_CONTENT}": response.completion,
+                            f"{LLM_OUTPUT_MESSAGES}.0.{MESSAGE_ROLE}": "assistant",
+                            OUTPUT_VALUE: response.completion,
```

Comment:
@RogerHYang Do we want to use message attributes for completion APIs? We don't use these attributes for OpenAI completions, but it does look nicer.

Reply:
I don't think we do that for openai. So maybe should follow the same precedent.